arrow/src/test/java/org/apache/calcite/adapter/arrow/ArrowAdapterTest.java

```diff
@@ -329,19 +329,10 @@ static void initializeArrowState(@TempDir Path sharedTempDir) throws IOException
     String sql = "select \"intField\", \"stringField\"\n"
         + "from arrowdata\n"
         + "where \"intField\" is null";
-    String plan;
-    if (Bug.CALCITE_6296_FIXED) {
-      plan = "ArrowToEnumerableConverter\n"
+    String plan = "ArrowToEnumerableConverter\n"
```

Comment:
Just a question, why not add a new test

Reply:
That test was already covering exactly this missing feature, and was meant to be fixed upon implementing it.
I think it's OK unless you see the need of a new test around `IS NULL` which is not covered here.

Reply:
Because this test involves two jira cases, and both are bug fixes, I think it should be tested separately and add jira links, which may help subsequent developers understand

Reply:
When you say it involves two Jira cases, you mean that it was added in one past ticket and that we are modifying it in the context of this one?

The ticket has been originally inserted as disabled (per my request), I have subsequently enabled it when the fallback on `Enumerable` was added, with the guard on the present ticket exactly to have the test fixed as in this PR.

The value of the previous version of test was to exercise the fallback while the feature wasn't supported, but now that part is not needed anymore and it can be removed.

For clarity we can add a link to the present Jira ticket as it fully and exclusively covers the feature, and the old tickets don't apply anymore.

Would that be OK for you, @caicancai?

Reply:
@timgrein please check https://github.com/apache/calcite/pull/3858/files#r1680614476 for the comment format